src/Gaffer/SplinePlug.cpp

```diff
@@ -360,6 +360,22 @@ void SplinePlug<T>::setToDefault()
 	setValue( m_defaultValue );
 }
 
+template<typename T>
+void SplinePlug<T>::resetDefault()
+{
+	const T newDefault = getValue();
+	const T oldDefault = m_defaultValue;
+	Action::enact(
+		this,
+		[this, newDefault] () {
+			this->m_defaultValue = newDefault;
```

Comment:
This was a bit of an odd one, and I wasn't sure myself. I added it in ValuePlug because there _are_ some nodes that use `setToDefault()`or `defaultValue()` in a compute, even though I suspect they won't face changing defaults.

SplinePlug is a stranger case for a few reasons :

- It's not a leaf level plug, so we wouldn't propagate dirtiness for it directly, but rather for all its leaf-level descendants (the component plugs of the points/colors).
- There's no guarantee that the number of points in the default is matched by the number of plugs actually present, so there's a bit of a mismatch in terms of what `resetDefault()` does anyway.

Because the mapping is much less one-to-one and because I doubted anyone would use `SplinePlug::defaultValue()` like that, I hesitated and didn't add the propagation. It would be trivial to do using `RecursiveRange` though, if you think I should.

Reply:
I suppose the only mechanism I have for attempting to make a call on that one would be doing it for consistencies sake? Is there a downside?

Reply:
I've added it in ebf4df5. Not sure if that's the right call, but we'll find out...

Reply:
Thanks @johnhaddon, me neither...